dive_core/event_state.h

```diff
@@ -8076,6 +8398,312 @@ template<typename CONFIG> class EventStateInfoT
                "attachment";
     }
 
+    //-----------------------------------------------
+    // FIELD ResolveScissor: Defines the rectangular bounds of the resolve operation
+
+    // `ResolveScissorPtr()` returns a shared pointer to an array of `size()` elements
+    inline const VkRect2D* ResolveScissorPtr() const
+    {
+        return reinterpret_cast<VkRect2D*>(reinterpret_cast<uint8_t*>(m_buffer.get()) +
```

Comment:
nit: Use `std::byte` instead of `uint8_t`.

It might be worth while to have a 
```cpp
template<typename T>
T* ResolvePtrAt(size_t offset) {
    // Maybe do a pointer bound check for debug?
    return reinterpret_cast<T*>(reinterpret_cast<std::byte*>(m_buffer.get())+offset);
}
```